File: Python/8py 1/find_average.py
"""
|--------------------------------------------------------------------------
| Calculate average
|--------------------------------------------------------------------------
|
"""

def find_average(numbers):
   return sum(numbers) / len(numbers)


from numpy import mean as find_average


def find_average(a):
   return sum(a)/len(a) if a else 0


from statistics import mean
def find_average(numbers):
   return mean(numbers)


print(find_average([1, 2, 3]), 2)

File: PHP/8php 1/find_average.php
<?php
/*
|--------------------------------------------------------------------------
| Calculate average
|--------------------------------------------------------------------------
|
*/

function find_average($array) {
   return count($array) == 0 ? 0 : array_sum($array) / count($array);
}


function find_average($a) {
   $s = 0;
   foreach($a as $i){
      $s += $i;
   }
   return $s / count($a);
}


function find_average($array) {
   return array_sum($array) / count($array);
}


function find_average($array) {
   $sum = 0;
   $count = 0;
   foreach ( $array as $item ) {
      $sum += $item;
      $count++;
   }
   return $sum / $count;
}


print_r(find_average([1, 1, 1]). ' ' . 1 . '
');
print_r(find_average([]). ' ' . 0 . '
');
print_r(find_average([1, 2, 3]). ' ' . 2 . '
');
print_r(find_average([1, 2, 3, 4]). ' ' . 2.5 . '
');